src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -146,6 +151,54 @@ public PersistentDataStore() {
 
     return messages;
   }
+  /*
+  * Loads all Event objects from the DataStore service and returns them in a List.
+  *
+  * @throws PersistentDataStoreException if an error was detected during the load from the
+  * DataStore service
+  */
+
+  public List<Event> loadEvents() throws PersistentDataStoreException {
+    List<Event> events = new ArrayList<>();
+    Query query = new Query("chat-events");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity: results.asIterable()) {
+      try {
+        // Loads in each variable
+        Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
+        String eventType = (String) entity.getProperty("event_type");
+        String userName = (String) entity.getProperty("username");
+        String userLink = (String) entity.getProperty("userlink");
+        String conversationName = (String) entity.getProperty("conversation-name");
+        String conversationLink = (String) entity.getProperty("conversation-link");
+        boolean inOrOut = (boolean) entity.getProperty("inOrOut");
+        // Depending on what type of event type it is; add different types of subclasses to event list.
+        if (eventType == "login-event") {
```

Comment:
having a separate Event sub-class for each event type works, but it is unscalable. It makes adding new event types non-trivial and you might end up having a LOT of sub-classes that'd be hard to maintain and work with. A generic Event class with a "type" field to differentiate between types would be a cleaner approach.
For the current feature, this approach works fine , so don't bother changing it :).